submission/loglist_refresher.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package submission
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/x509util"
+)
+
+const (
+	// LoglistRefreshInterval is interval between consecutive external reads of Log-list.
+	loglistRefreshInterval = time.Hour * 24 * 7
+)
+
+// LogListEvent wraps result of sinle Log list refresh.
+type LogListEvent struct {
+	Ll  *loglist.LogList
+	Err error
+}
+
+// LoglistRefresher regularly reads Log-list and emits notifications when updates/errors
+// observed.
+type LoglistRefresher struct {
+	mu   sync.RWMutex
+	ll   *loglist.LogList
+	path string
+
+	Events chan *LogListEvent
```

Comment:
You need to document this field, since you're exporting it. It's also conventional to put exported fields above unexported fields.

Reply:
Done.